src/Data/Permissions.php

```diff
@@ -0,0 +1,41 @@
+<?php
+namespace App\Data;
+
+use App\Data\Entity\User;
+
+class Permissions {
+
+    /**
+     * @var Authenticator
+     */
+    protected $auth;
+
+    /**
+     * @var User
+     */
+    protected $user;
+
+    public function __construct(Authenticator $auth)
+    {
+        $this->auth = $auth;
+    }
+
+    public function setUser(User $user)
```

Comment:
Sharing the current user seems like a reasonable thing to do. User context should be consistent across the lifecycle of a request in 99% of cases.